From: Simon McVittie <smcv@debian.org>
Date: Wed, 17 Jul 2019 13:54:41 +0100
Subject: gdk: Don't distribute generated files in tarballs

gdkversionmacros.h is generated by configure, so it should be in
DISTCLEANFILES. The rest of $(gdk_built_sources) are built at compile
time by GLib tools, so in principle they should be in CLEANFILES, but
DISTCLEANFILES is close enough.

This is a continuation of the same general topic as commit dad773b1.

Signed-off-by: Simon McVittie <smcv@debian.org>
Forwarded: https://gitlab.gnome.org/GNOME/gtk/merge_requests/1001
Bug: https://gitlab.gnome.org/GNOME/gtk/issues/357
---
 gdk/Makefile.am | 29 ++++++-----------------------
 1 file changed, 6 insertions(+), 23 deletions(-)

diff --git a/gdk/Makefile.am b/gdk/Makefile.am
index 710a548..fbdd193 100644
--- a/gdk/Makefile.am
+++ b/gdk/Makefile.am
@@ -203,16 +203,8 @@ nodist_gdkinclude_HEADERS = gdkconfig.h gdkenumtypes.h gdkversionmacros.h
 deprecatedincludedir = $(includedir)/gtk-3.0/gdk/deprecated
 deprecatedinclude_HEADERS = $(deprecated_h_sources)
 
-common_sources = 		\
-	$(gdk_private_headers)	\
-	$(gdk_c_sources)	\
-	gdkenumtypes.c		\
-	gdkmarshalers.c		\
-	gdkmarshalers.h		\
-	gdkresources.h		\
-	gdkresources.c
-
-libgdk_3_la_SOURCES = $(common_sources)
+libgdk_3_la_SOURCES = $(gdk_private_headers) $(gdk_c_sources)
+nodist_libgdk_3_la_SOURCES = $(gdk_built_sources)
 libgdk_3_la_CFLAGS = $(AM_CFLAGS) $(GDK_HIDDEN_VISIBILITY_CFLAGS)
 libgdk_3_la_LIBADD = $(GDK_DEP_LIBS) $(SHM_LIBS)
 libgdk_3_la_LDFLAGS = $(LDADD)
@@ -412,10 +404,8 @@ endif
 
 lib_LTLIBRARIES = libgdk-3.la
 
-MAINTAINERCLEANFILES = $(gdk_built_sources) stamp-gdkenumtypes.h
-EXTRA_DIST += \
-	$(gdk_built_sources)	\
-	fallback-c89.c
+DISTCLEANFILES = $(gdk_built_sources) stamp-gdkenumtypes.h
+EXTRA_DIST += fallback-c89.c
 
 install-exec-hook:
 if DISABLE_EXPLICIT_DEPS
@@ -499,7 +489,7 @@ gdkresources.c: gdk.gresource.xml $(resource_files)
 # ------------------- MSVC Build Items ----------------
 MSVCPROJS = gdk-3
 
-gdk_3_FILES = $(libgdk_3_la_SOURCES)
+gdk_3_FILES = $(libgdk_3_la_SOURCES) $(nodist_libgdk_3_la_SOURCES)
 gdk_3_EXCLUDES = dummy
 gdk_3_HEADERS_DIR = $(gdkincludedir)
 
@@ -553,20 +543,13 @@ dist-hook: \
 	$(top_builddir)/win32/vs9/gdk-3.headers	\
 	$(INTROSPECTION_INTERMEDIATE_ITEMS)
 
-DISTCLEANFILES = gdkconfig.h stamp-gc-h
+DISTCLEANFILES += gdkconfig.h stamp-gc-h
 
 install-data-local: install-ms-lib install-def-file
 
 uninstall-local: uninstall-ms-lib uninstall-def-file
 	rm -f $(DESTDIR)$(configexecincludedir)/gdkconfig.h
 
-# if srcdir!=builddir, clean out maintainer-clean files from builddir
-# this allows dist to pass.
-distclean-local:
-	if test $(srcdir) != .; then \
-	  rm -f $(MAINTAINERCLEANFILES); \
-	fi
-
 .PHONY: files
 
 files:
